kanidmd/src/lib/constants/acp.rs

```diff
@@ -126,6 +126,29 @@ pub const JSON_IDM_SELF_ACP_WRITE_V1: &str = r#"{
     }
 }"#;
 
+pub const JSON_IDM_PEOPLE_SELF_ACP_WRITE_MAIL_V1: &str = r#"{
+    "attrs": {
+        "class": ["object", "access_control_profile", "access_control_modify"],
+        "name": ["idm_people_self_acp_write_mail"],
+        "uuid": ["00000000-0000-0000-0000-ffffff000041"],
+        "description": ["Builtin IDM Control for self write of mail for people accounts."],
+        "acp_receiver": [
+            "{\"eq\":[\"memberof\",\"00000000-0000-0000-0000-000000000033\"]}"
+        ],
+        "acp_targetscope": [
+        ],
+        "acp_targetscope": [
+            "{\"and\": [{\"eq\": [\"class\",\"person\"]}, \"self\"]}"
```

Comment:
I think it should be :

```
"{\"and\": [{\"eq\": [\"class\",\"person\"]}, {\"eq\": [\"class\",\"account\"]} , \"self\"]}"
```

Since a person can exist as a phone-book style entry, but can't authenticate, so only accounts (can auth) and people (are human) can update mail.

Reply:
Sure, updated to that. That makes some sense.

It does seem like it might be inconsistent with the ACP above though, `idm_self_acp_write`. That one has `acp_targetscope: ["self"]`, and allows modifying `legalname`, among other things, with no requirement of `class: "account"` (or not that I see anyway?)

With my current understanding, I also don't see how anything but an account could auth and thus be subject to this ACP.

Is this change to prevent some actual thing now, or does it fit the model better, but in practice not have an observable difference now?
If these phone-book style "people" entries existed (or do they?), would the ACP be able to apply to them in some way?

Reply:
Actually, we should fix that too then 😅

Reply:
Sure, done, pushed as an additional commit to this branch.

It would be nice to write a unit test too if it's possible to create a non-account-class "self" that can also call the API.. which I guess is what I'm asking above already. I'm happy to wait for an answer there, or do more digging through code/docs if you've got pointers.

Reply:
I don't think you can, not without some of the internal server magic methods. The server internally has the ability to impersonate *any* entry of the system, which allows testing some of these properties, but they aren't exposed (yet) to the API as the security properties and limits on that need clearer definition and design. There are already self write tests, so lets just leave them as is :)